### PR TITLE
feat(ui): ability to update source settings

### PR DESCRIPTION
**Pull request by martinmaillard**

### CODE REVIEW

File: ui/src/components/project/SourceForm.vue [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  @Prop() save: (source: SourceFormData) => void;
+  data: SourceFormData;
+
+  created () {
+    this.data = {
+      id: this.source.id,
+      name: this.source.name,
+      csvDelimiter: this.source.csvDelimiter,
+      csvQuote: this.source.csvQuote,
+    }
```
Comment (tpluscode) [inline ui/src/components/project/SourceForm.vue]:
```js
this.data = { ...this.source }
```

Maybe like this ?

File: ui/src/store/modules/projects.ts [hunk 2]
```diff
@@ -66,14 +66,6 @@ const actions: ActionTree<ProjectsState, RootState> = {
...
     })
   },
-
-  async loadSources (context, project) {
-    await handleAPIError(context, async () => {
-      const sources = await client.projects.getSources(project)
-
-      context.commit('storeSources', { project, sources })
-    })
-  },
```
Comment (tpluscode) [inline ui/src/store/modules/projects.ts]: Where did this go?

File: ui/src/components/project/SourceForm.vue (hunk 1 above)
Comment (martinmaillard) [reply ui/src/components/project/SourceForm.vue]: I could, but then my object would have more keys that I don't need. It doesn't make a technical difference, but I think it makes things more confusing. Arguable I guess :)

File: ui/src/store/modules/projects.ts (hunk 2 above)
Comment (martinmaillard) [reply ui/src/store/modules/projects.ts]: It went in `sources/loadForProject`. It's a remain of a previous refactoring.

Comment (martinmaillard) [reply ui/src/store/modules/projects.ts]: It went in sources/loadForProject. It's a remain of a previous refactoring.

**Pull request merged**